scripts/release.sh

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "enter release string according to semantic versioning (e.g. v1.2.3)."
+read -r INPUT
+if [[ ! "${INPUT}" =~ ^v[0-9]+.[0-9]+.[0-9]+ ]]; then
+  echo "Expected 'version' param of the form 'v<major-version>.<minor-version>.<patch-version>' but got '${INPUT}'"
+  exit 1
+fi
+
+VERSION=${INPUT#v}
+RELEASE_VERSION="${VERSION}"
+MINOR_VERSION=$(echo "${VERSION}" | cut -d. -f 1-2)
+
+REPOSITORY=${REPOSITORY:-"git@github.com:etcd-io/bbolt.git"}
+REMOTE="${REMOTE:-"origin"}"
+
+# ensuring the minor-version is identical.
+source_version=$(grep -E "\s+Version\s*=" ./version/version.go | sed -e "s/.*\"\(.*\)\".*/\1/g")
+if [[ "${source_version}" != "${RELEASE_VERSION}" ]]; then
+   source_minor_version=$(echo "${source_version}" | cut -d. -f 1-2)
+   if [[ "${source_minor_version}" != "${MINOR_VERSION}" ]]; then
+     echo "Wrong bbolt minor version in version.go. Expected ${MINOR_VERSION} but got ${source_minor_version}. Aborting."
+     exit 1
+   fi
+fi
+
+# creating a branch to bump 'version.go'.
+date_string=$(date +%Y%m%d)
+local_branch_name="version_${date_string}"
+local_branch_err=$(git checkout -b "${local_branch_name}" | grep -E "error|fatal" || true )
+if [[ -n "${local_branch_err}" ]]; then
+  echo "${local_branch_err}"
+fi
+
+# bump 'version.go'.
+echo "Updating version from '${source_version}' to '${RELEASE_VERSION}' in 'version.go'"
+sed -i "s/${source_version}/${RELEASE_VERSION}/g" ./version/version.go
+
+# push 'version.go' to remote.
+echo "committing 'version.go'"
+git add ./version/version.go
+git commit -s -m "Update version to ${VERSION}"
+git push -u -f "${REMOTE}" "${local_branch_name}"
+echo "'version.go' has been committed to remote repo."
+
+# create tag and push to remote.
+remote_tag_exists=$(git ls-remote --tags "${REPOSITORY}" | grep -c "${INPUT}" || true)
+if [ "${remote_tag_exists}" -gt 0 ]; then
+   echo "Release version tag exists on remote. Checking out refs/tags/${INPUT}"
+   git checkout -q "tags/${INPUT}"
+elif [ "${remote_tag_exists}" -eq 0 ]; then
+   echo "Creating new tag for '${INPUT}'"
+   git tag -f "${INPUT}"
```

Comment:
Do we want to GPG sign the tag in this repository, too? I don't think we've been doing it, but it may be easier with the script in place. What do you think, @ahrtr?

Reply:
> Do we want to GPG sign the tag in this repository, too?

I think so. It's exactly what I have been doing (manually).

Reply:
@ahrtr should the script

- pushes a branch with the changes and new tag to a fork (origin)
- pushes the changes directly to upstream (etcd-io) release branch (e.g. release-1.4)

Reply:
I think the latter, else the tag would be in a fork, and we'll need to manually tag it later.

Reply:
A gentle reminder, @ahrtr. I think we're just waiting for confirmation on this.

Reply:
You just need to test in your fork by setting `REPOSITORY` to your fork, i.e. `git@github.com:Elbehery/bbolt.git`, and make sure that `REMOTE` points to how the remote is named in your local repository. In my case, `origin`: etcd-io, `ivanvc`: my fork. YMMV

Reply:
updated 👍🏽 ptal